src/main/java/ro/utcn/sp1tz3/Assignment1/controller/ConsoleController.java

```diff
@@ -0,0 +1,254 @@
+import ro.utcn.sp1tz3.Assignment1.service.TagManagementService;
+import ro.utcn.sp1tz3.Assignment1.service.UserManagementService;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class ConsoleController implements CommandLineRunner {
+    private final Scanner scanner = new Scanner(System.in);
+    private final UserManagementService userManagementService;
+    private final QuestionManagementService questionManagementService;
+    private final AnswerManagementsService answerManagementsService;
+    private final TagManagementService tagManagementService;
+    private boolean log = false;
+    private User currentUser;
+
+    @Override
+    public void run(String... args){
+        System.out.println("welcome to StackOverflow");
+        boolean done = false;
+        while(!done){
+            System.out.println("Enter a command: ");
+            String command = scanner.next().trim();
+            try{
+                done = handleCommand(command);
+            } catch (UserNotFoundException userNotFoundException){
+                System.out.println("The user with the given ID was not found");
+            }
+        }
+    }
+
+    private boolean handleCommand(String command){
+        switch(command){
+            case "register":
+                handleRegister();
+                return false;
+            case "login":
+                handleLogin();
+                return false;
+            case "logoff":
+                handleLogoff();
+                return false;
+            case "list":
+                handleList();
+                return false;
+            case "listtags":
+                handleListTags();
+                return false;
+            case "listusers":
+                handleListUsers();
+                return false;
+            case "ask":
+                handleAddQuestion();
+                return false;
+            case "addtag":
+                handleAddTag();
+                return false;
+            case "filter":
+                handleFilterQuestions();
+                return false;
+            case "answer":
+                handleAddAnswer();
+                return false;
+            case "remove":
+                handleRemove();
+                return false;
+            case "exit":
+                return true;
+            default:
+                System.out.println("Unknown command. Try again");
+                return false;
+        }
+    }
+
+    private void handleRegister(){
+        System.out.println("Username");
+        String username = scanner.next().trim();
+        System.out.println("Password");
+        String password = scanner.next().trim();
+        List<User> users = userManagementService.listUsers();
+        boolean taken = false;
+        for(User s: users){
```

Comment:
All this business logic from the controller class should be inside the service layer. The fact that two users cannot have the same name is a *business rule*.